inspirehep/modules/records/checkers.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2018 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Records checkers."""
+
+from __future__ import absolute_import, division, print_function
+
+from collections import defaultdict
+
+from sqlalchemy import type_coerce
+from sqlalchemy.dialects.postgresql import JSONB
+
+from invenio_records.models import RecordMetadata
+from inspire_utils.record import get_value
+
+
+def increase_cited_count(result, identifier, core):
+    """Increases the number of times a reference with the same identifier has appeared"""
+    if core:
+        result[identifier] = (result[identifier][0] + 1, result[identifier][1])
+    else:
+        result[identifier] = (result[identifier][0], result[identifier][1] + 1)
+
+
+def calculate_score_of_reference(counted_reference):
+    """Given a tuple of the number of times cited by a core record and a non core record,
+    calculate a score associated with a reference.
+
+    The score is calculated giving five times more importance to core records"""
+    _, (core_count, non_core_count) = counted_reference
+    return core_count * 5 + non_core_count
+
+
+def order_dictionary_into_list(result_dict):
+    """Return ``result_dict`` as an ordered list of tuples"""
+    sorted_list = sorted(result_dict.items(), key=calculate_score_of_reference, reverse=True)
+
+    return sorted_list
+
+
+def get_all_unlinked_references():
+    """Return a list of dict, in which each dictionary corresponds to one reference object
+    and the status of core or non core"""
+    query = RecordMetadata.query.filter(type_coerce(RecordMetadata.json, JSONB)['_collections'].contains(['Literature'])).with_entities(RecordMetadata.json)
```

Comment:
can you split this line, it's a bit too long? you should try not to go too much above 100 as it becomes hard to read afterwards. One nice way to split it (that we don't currently use in inspire, but should IMHO) is to surround it with `()` as here: https://github.com/ambv/black#call-chains.